pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -84,9 +88,23 @@ func (r *AtlasProjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	return ctrl.Result{}, nil
 }
 
+func (r *AtlasProjectReconciler) Delete(obj runtime.Object) error {
+	// TODO CLOUDP-80607
+	return nil
+}
+
 func (r *AtlasProjectReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&mdbv1.AtlasProject{}).
+		Watches(
+			&source.Kind{Type: &mdbv1.AtlasCluster{}},
+			&watch.DeleteEventHandler{Controller: r},
+			builder.WithPredicates(watch.DeleteOnly()),
+		).
+		Watches(
```

Comment:
we register the new watch for Secrets - so any events on Secrets will call the resource handler